<launch>
  <param name="/use_sim_time" value="true" />
  <arg name="robot_name" value="robot8"/>
  <env name="robot_name" value="$(arg robot_name)"/>
  <param name="tf_prefix" value="$(arg robot_name)" />
  <group ns="$(arg robot_name)">
    <node name="playbag" pkg="rosbag" type="play"
      args= "
            /home/tristan/Desktop/2021-11-26-14-32-07.bag 
            --topics /robot8/scan /robot8/raw_odom /robot8/imu/data /robot8/imu/mag 
            --clock
            " 
    />
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_footprint_to_base_link" args="0 0 0.115 0 0 0  /$(arg robot_name)/base_footprint /$(arg robot_name)/base_link"/>
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_laser" args="-0.1 0 0.035 4.712 0 0  /$(arg robot_name)/base_link /$(arg robot_name)/laser" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="base_link_to_imu_link" args="0 0.05 -0.015 0 0 3.1415  /$(arg robot_name)/base_link /$(arg robot_name)/imu_link"/>
    
    <include file="$(find languageslam)/launch/linorobot/laser_conf_l.launch" >
        <arg name="robot_name"  value="$(arg robot_name)" />
    </include>


    <!--node name="cartographer_node" pkg="cartographer_ros"
        type="cartographer_node" args="
            -configuration_directory $(find languageslam)/config
            -configuration_basename client.lua"
        output="screen">
        <remap from="/scan" to="scan" />
        <remap from="/odom" to="odom" />
    </node>

    <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
        type="cartographer_occupancy_grid_node" args="-resolution 0.05" /-->
    


    <node pkg="robot_localization" type="ekf_localization_node" name="ekf_localization" output="screen">
        <remap from="odometry/filtered" to="odom" />
        <rosparam command="load" file="$(find languageslam)/config/linorobot/robot_localization.yaml" subst_value="true" />
    </node>
  </group>
</launch>